src/main/java/org/mskcc/cbio/oncokb/web/rest/SlackController.java

```diff
@@ -92,12 +94,31 @@ public ResponseEntity<String> approveUser(@RequestParam("payload") String action
                                     userDTO = updatedUserDTO.get();
                                 }
                             }
+                        
+                            boolean apiAccessRequested = userDTO.getAdditionalInfo() != null && userDTO.getAdditionalInfo().getApiAccessRequest() != null && userDTO.getAdditionalInfo().getApiAccessRequest().isRequested();
+                            if (userDTO.getLicenseType() != LicenseType.ACADEMIC || apiAccessRequested) {
+                                Set<String> userDTOAuthorities = userDTO.getAuthorities();
+                                userDTOAuthorities.add(AuthoritiesConstants.API);
+                            }
+
                             Optional<UserDTO> updatedUser = userService.approveUser(userDTO, false);
                             if (updatedUser.isPresent() && updatedUser.get().isActivated()) {
                                 mailService.sendApprovalEmail(userDTO);
                             }
                         }
                         break;
+                    case APPROVE_USER_FOR_API_ACCESS:
+                        boolean apiAccessRequested = userDTO.getAdditionalInfo() != null && userDTO.getAdditionalInfo().getApiAccessRequest() != null && userDTO.getAdditionalInfo().getApiAccessRequest().isRequested();
+                        if (userDTO.getLicenseType() != LicenseType.ACADEMIC || apiAccessRequested) {
+                            Set<String> userDTOAuthorities = userDTO.getAuthorities();
+                            userDTOAuthorities.add(AuthoritiesConstants.API);
+                        }
```

Comment:
We don't need to check the request anymore here, do we?

Reply:
Yeah there should always be an API access request and the boolean isRequested should always be set. Can additional info ever be null?

Reply:
Nvm, it's a good point. Referring to my the other comment, I think it's better keep the checking here. However it's the exact same logic of line 98-102. Can we consolidate and create a method?